packages/web3-providers-http/src/index.ts

```diff
@@ -1,17 +1,20 @@
+import fetch from 'cross-fetch';
 import {
+	EthExecutionAPI,
+	InvalidClientError,
+	JsonRpcResponse,
+	MethodNotImplementedError,
 	ResponseError,
-	JsonRpcPayload,
+	Web3APIMethod,
+	Web3APIPayload,
+	Web3APIReturnType,
+	Web3APISpec,
 	Web3BaseProvider,
-	MethodNotImplementedError,
-	JsonRpcResponse,
 	Web3BaseProviderStatus,
-	InvalidClientError,
 } from 'web3-common';
-import fetch from 'cross-fetch';
-
 import { HttpProviderOptions } from './types';
 
-export class HttpProvider extends Web3BaseProvider {
+export class HttpProvider<API extends Web3APISpec = EthExecutionAPI> extends Web3BaseProvider<API> {
```

Comment:
same here, change in design request, we shouldn't enforce provider as only Eth Provider.